models/DataObject/ClassDefinition/Helper/OptionsProviderResolver.php

```diff
@@ -32,9 +32,21 @@ class OptionsProviderResolver extends ClassResolver
 
     public static function resolveProvider(?string $providerClass, int $mode): ?object
     {
+
         return self::resolve($providerClass, function ($provider) use ($mode) {
+            if ($provider instanceof MultiSelectOptionsProviderInterface) {
+                trigger_deprecation(
+                    'pimcore/pimcore',
+                    '12.0',
+                    sprintf('Implementing %s is deprecated, use %s instead', MultiSelectOptionsProviderInterface::class, SelectOptionsProviderInterface::class)
+                );
```

Comment:
The deprecation will be introduced in Pimcore 11.1. Also, there's no need for `sprintf`.

```suggestion
                trigger_deprecation(
                    'pimcore/pimcore',
                    '11.1',
                    'Implementing %s is deprecated, use %s instead',
                    MultiSelectOptionsProviderInterface::class,
                    SelectOptionsProviderInterface::class,
                );
```